File: src/ssr_pkg/src/yao_node.cpp
#include <ros/ros.h>
#include <std_msgs/String.h>

int main(int argc, char *argv[])
{
    ros::init(argc, argv, "yao_node");

    // nh 负责管理话题创建，消息发送
    ros::NodeHandle nh;
    // 规定发布信道的主题名和信道容量，这里最多可以接收10个消息
    ros::Publisher pub = nh.advertise<std_msgs::String>("Topic2", 10);
    ros:: Rate loop_rate(10);

    while(ros::ok()) {
        printf("yao ok\n");
        // 初始化消息对象
        std_msgs::String msg;
        // 消息内容
        msg.data = "飞+++";
        // 发送消息
        pub.publish(msg);
        loop_rate.sleep();
    }
    return 0;
}

File: src/ssr_pkg/scripts/yao_node.py
#!/usr/bin/env python3
#coding=utf-8

import rospy
from std_msgs.msg import String

if __name__ == "__main__":
    rospy.init_node("yao_node")
    rospy.logwarn("yao ok!")
    # 新建发布对象，指定消息标题，消息包内容类型，消息包容量
    pub = rospy.Publisher("Topic2", String, queue_size=10)
    # 发布消息频率为10HZ
    rate = rospy.Rate(10)
    
    while not rospy.is_shutdown():
        rospy.loginfo("yao going!")
        msg = String()
        msg.data = "yao fly!"
        pub.publish(msg)
        rate.sleep()